src/controller/postUrl.ts

```diff
@@ -0,0 +1,43 @@
+import express from "express";
+import { databaseModel } from "../model";
+import { nanoid } from "nanoid";
+import { logger } from "../services";
+import { response } from "../utils";
+
+export const postUrl = async (req: express.Request, res: express.Response) => {
+  try {
+    const { fullUrl, customUrl } = req.body;
+
+    logger.info("Request body:", req.body);
+
+    if (!fullUrl) {
+      return res.status(400).json(response.error("fullUrl is required."));
+    }
+
+    const checkUrl = await databaseModel.findOne({ fullUrl });
+    if (checkUrl) {
+      return res.status(409).json(response.linkExist(checkUrl));
+    }
+
+    if (customUrl) {
+      const checkCustomName = await databaseModel.findOne({
+        shortUrl: customUrl,
+      });
+      if (checkCustomName) {
+        return res.status(409).json(response.customNameExist(checkCustomName));
+      }
+    }
+
+    const shortUrl = await databaseModel.create({
+      fullUrl,
+      shortUrl: customUrl || nanoid().substring(0, 10),
+    });
```

Comment:
- I can't find where you checked if the customName is greater than 5 characters (This check should be in the middleware)
- You should also extract this generation of customName to a util function